pkg/executor/simple.go

```diff
@@ -1816,7 +1834,18 @@ func (e *SimpleExec) executeAlterUser(ctx context.Context, s *ast.AlterUserStmt)
 			value any
 		}
 		var fields []alterField
+
 		if spec.AuthOpt != nil {
+			// Only use `REPLACE <pwd>` when changing the current user
```

Comment:
Should we reject a `SET PASSWORD` statement in `executeSetPwd` if the current user requires changing password?

Reply:
Good catch.

I've updated `SET PASSWORD` in 6def7e8df863d23541d43ad5a1fdbfc66900d7f3